mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/options/MongoDBCollection.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.mongodb.options;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import org.hibernate.ogm.datastore.mongodb.options.impl.MongoDBCollectionConverter;
+import org.hibernate.ogm.options.spi.MappingOption;
+
+/**
+ * MongoDB specific definitions for a collection.
+ *
+ * @author Guillaume Smet
+ */
+@Target({ TYPE, METHOD, FIELD })
+@Retention(RUNTIME)
+@MappingOption(MongoDBCollectionConverter.class)
+public @interface MongoDBCollection {
```

Comment:
What other options would that be? Me feeling is we should rather find options that are applicable to several stores. E.g. take `@TimeToLife`. Currently it's only there fore Redis, but it could be generalized and used for MongoDB, too. In your model `timeToLife()`, it could be a part of `@MongoDBCollection`, but I think it shouldn't so to provide a more unified experience when working with different stores. See where I'm coming from?